### PR TITLE
machine: Add retries for ssh

### PR DESCRIPTION
**Pull request by DaanDeMeyer**

### CODE REVIEW

File: mkosi/__init__.py [hunk 1]
```diff
@@ -8059,14 +8058,8 @@ def find_address(args: MkosiArgs) -> Tuple[str, str]:
...
-def run_command_image(args: MkosiArgs, commands: Sequence[str], timeout: int, check: bool, stdout: _FILE = sys.stdout, stderr: _FILE = sys.stderr) -> CompletedProcess:
-    if args.verb == Verb.qemu:
-        return run_ssh(args, commands, check, stdout, stderr, timeout)
-    elif args.verb == Verb.boot:
-        cmdline = ["systemd-run", "--quiet", "--wait", "--pipe", "-M", machine_name(args), "/usr/bin/env", *commands]
-        return run(cmdline, check=check, stdout=stdout, stderr=stderr, text=True, timeout=timeout)
-    else:
-        return run(run_shell_cmdline(args, pipe=True, commands=commands), check=check, stdout=stdout, stderr=stderr, text=True, timeout=timeout)
+def run_systemd_cmdline(args: MkosiArgs, commands: Sequence[str]) -> Sequence[str]:
+    return ["systemd-run", "--quiet", "--wait", "--pipe", "-M", machine_name(args), "/usr/bin/env", *commands]
```
Comment (behrmann) [inline mkosi/__init__.py:8062]: I never noticed this `/usr/bin/env` here. I'm not quite sure I understand what it's doing?

**Review by behrmann — COMMENTED** [review]
Nice. With this the Fedora CI job for ext4 finally finishes.

File: mkosi/__init__.py (hunk 1 above)
Comment (DaanDeMeyer) [reply mkosi/__init__.py:8062]: systemd-run creates a transient unit, and ExecStart= expected absolute paths until some recent systemd version, so we prefix commands with env to make sure binaries are looked up in the path.

File: mkosi/__init__.py [hunk 2]
```diff
@@ -8059,14 +8058,8 @@ def find_address(args: MkosiArgs) -> Tuple[str, str]:
...
 
-def run_command_image(args: MkosiArgs, commands: Sequence[str], timeout: int, check: bool, stdout: _FILE = sys.stdout, stderr: _FILE = sys.stderr) -> CompletedProcess:
-    if args.verb == Verb.qemu:
-        return run_ssh(args, commands, check, stdout, stderr, timeout)
-    elif args.verb == Verb.boot:
-        cmdline = ["systemd-run", "--quiet", "--wait", "--pipe", "-M", machine_name(args), "/usr/bin/env", *commands]
-        return run(cmdline, check=check, stdout=stdout, stderr=stderr, text=True, timeout=timeout)
-    else:
-        return run(run_shell_cmdline(args, pipe=True, commands=commands), check=check, stdout=stdout, stderr=stderr, text=True, timeout=timeout)
+def run_systemd_cmdline(args: MkosiArgs, commands: Sequence[str]) -> Sequence[str]:
```
[the remark this answers is not in the document]
Comment (DaanDeMeyer) [reply mkosi/__init__.py:8061]: I'm not really convinced., I agree it's not ideal though but it'll do the job

**Comment by DaanDeMeyer** [comment]
Going to merge this one since there were only trivial changes and I want to update all the other PRs with the CI fixes. If there's any remaining issues I can do a follow up PR

**Pull request merged**